pkg/controller/core.oam.dev/v1alpha2/application/dispatch/dispatch.go

```diff
@@ -116,15 +116,49 @@ func (a *AppManifestsDispatcher) Dispatch(ctx context.Context, manifests []*unst
 	return a.currentRT.DeepCopy(), nil
 }
 
-// ReferenceScopes add workload reference to scopes' workloadRefPath
-func (a *AppManifestsDispatcher) ReferenceScopes(ctx context.Context, wlRef *v1.ObjectReference, scopes []*v1.ObjectReference) error {
-	// TODO handle scopes
+// ReferenceScopes add workload reference to scopes' workload reference field
+func (a *AppManifestsDispatcher) ReferenceScopes(ctx context.Context, scopeRefs map[v1.ObjectReference]v1alpha2.CompReferences) error {
+	appName := ExtractAppName(a.currentRTName, a.namespace)
+	for s, compRefs := range scopeRefs {
+		scopeObj := &unstructured.Unstructured{}
+		scopeObj.SetAPIVersion(s.APIVersion)
+		scopeObj.SetKind(s.Kind)
+		if err := a.c.Get(ctx, client.ObjectKey{Name: s.Name, Namespace: a.namespace}, scopeObj); err != nil {
+			return err
+		}
+		copyS := scopeObj.DeepCopy()
+
+		tmp := map[string]interface{}{}
+		for k, v := range compRefs {
+			tmp[k] = v
+		}
+		// TODO hard-code reference field path should be replcaced with getting from scope definition
+		if err := unstructured.SetNestedMap(scopeObj.Object, tmp, "spec", "appReferences", appName); err != nil {
```

Comment:
@wonderflow call for help
As shown here, I hope to set ComponentReferences to an unsturctued object's specific field(spec.appReferences), but it failed for "Cannot deep copy ComponentReferences".

Reply:
solved.